pgml-docs/docs/blog/benchmarking.md

```diff
@@ -0,0 +1,207 @@
+(2 rows)
+```
+
+Benchmarks
+----------
+
+### SQL
+```sql
+CREATE OR REPLACE FUNCTION sql_norm_l2(vector REAL[]) 
+  	RETURNS REAL
+	LANGUAGE sql
+	LEAKPROOF
+	IMMUTABLE
+	STRICT
+	PARALLEL SAFE
+AS $$
+  SELECT SQRT(SUM(squared.values))
+  FROM (SELECT UNNEST(vector) * UNNEST(vector) AS values) AS squared;
+$$;
+```
+
+```sql
+SELECT sql_norm_l2(vector) FROM embeddings;
+```
+110ms
+
+### PLPGSQL
+```sql
+CREATE OR REPLACE FUNCTION plpgsql_norm_l2(vector REAL[]) 
+  	RETURNS REAL
+	LANGUAGE plpgsql
+	LEAKPROOF
+	IMMUTABLE
+	STRICT
+	PARALLEL SAFE
+AS $$
+	BEGIN
+		RETURN SQRT(SUM(squared.values))
+		FROM (SELECT UNNEST(vector) * UNNEST(vector) AS values) AS squared;
+	END
+$$;
+```
+
+```sql
+SELECT plpgsql_norm_l2(vector) FROM embeddings;
+```
+120 ms
```

Comment:
Try switching to sql/output formatting we use elsewhere and maybe have the output of \timing there as well? In bold or something. Could look good.